contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checker.py

```diff
@@ -175,17 +173,42 @@ def execute(self):
     # If we are linting for Python 3, skip lint altogether.
     # Long-term Python 3 linting solution tracked by:
     # https://github.com/pantsbuild/pants/issues/5764
-    interpreter = self.context.products.get_data(PythonInterpreter)
-    # Check interpreter is not 'None' for test cases that do not
-    # run the python interpreter selection task.
-    if interpreter and interpreter.version >= (3, 0 ,0):
-      self.context.log.info('Linting is currently disabled for Python 3 targets.\n '
+    python_tgts = self.context.targets(
+      lambda tgt: isinstance(tgt, (PythonTarget))
+    )
+    if not python_tgts:
+      return
+    interpreter_cache = PythonInterpreterCache(PythonSetup.global_instance(),
```

Comment:
`PythonSetup` and `PythonRepos` are `subsystem_dependencies` you need to declare above.